### ISSUE TITLE
Зміна структури Excursion Type для підтримки генерації турів.

### ISSUE THREAD

**Issue by msenyk** [msenyk/sf-practicum-2022#7]
- Визначити нові поля (узгодити)
- Додати поля на Page Layout
- Додати правила валідації

Поля, які я пропоную додати:
- Number of tickets (Number)
- First day of excursion (Date/Time)
- Last day of excursion (Date/Time)
- Interval (Number)

Додано поля:
- Number of tickets (Number)
- Days of excursion Picklist (Multi-Select)
- Schedualed (Checkbox)
- Time of beginning (Time)

Додано правила валідації:

- NumberOfTicketsNotBelowZero (_Number of tickets cant be less then zero_)
- IfSchedualedShouldHaveDays (_The excursion is schedualed, it should have days_)
- DaysIfNotSchedualed (_The excrusion is not scedualed, it should have any days_)

**Comment by msenyk** [comment]
@thxnk Погоджуюсь із Number of tickets (Number).
Для First/Last day - прийдеться оновлювати щоразу, коли генеруємо тури. Тому їх немає сенсу тримати в базі.
Якщо я хочу зробити тур кожної п'ятниці, то яким має бути Interval? А якщо в четвер та вівторок?

**Comment by msenyk** [comment]
@thxnk Давайте орієнтуватися на можливість визначити генерацію подій по днях тижня.

**Comment by thxnk** [comment]
Тоді додам тільки Number of tickets.
А дати початку і кінця вже при додаванні буде вибирати користувач.
Щодо днів неділі, то додам Picklist (Multi-Select).

**Comment by msenyk** [comment]
Так, ще варто додати поле Checkbox яке буде визначати, що цей тип екскурсій для створення турів за розкладом. Бо є екскурсії, які проводять лише час від часу і механізм генерації має їх не чіпати.

**Comment by thxnk** [comment]
Добре, зрозумів. Тоді почну виконувати завдання далі.

**Comment by msenyk** [comment]
І ще треба час початку екскурсії.

**Comment by thxnk** [comment]
Ще є одне питання. Треба добавляти picklist саме до об'єкту, чи вже краще на компоненті для генерації туру? Якщо зробити цей picklist саме на компоненті, то тоді для одного типу екскурсій можна буде не змінюючі сам об'єкт генерувати тури на різні дні. Це виглядає набагато зручніше.

**Comment by msenyk** [comment]
Це як подивитись на задачу. Якщо в нас екскурсія завджи по вихідних днях, то простіше обрати то на об'єкті (записі) один раз і потім лише генерувати кожного місяця. А якщо ці дні часто міняються, то дійсно краще мати на компоненті, але тоді збільшується ризик помилки, що ми забудемо вказати правильні дні.
Давайте зараз зробимо поле на рівні об'єкту.

**Comment by msenyk** [comment]
@thxnk Деякі зауваження:
- виправити назву поля Schedualed --> Is Scheduled
- змінити назву поля Days of excursion --> Days of Week
- API Name значень пікліста має бути вказані англійською, тобто Sunday, Monday тощо
- додати окрему секцію Timetable на page layout та перемістити туди всі нові поля
- виправити валідацію, щоб не дозволяла зберігати кількість квитків 0. Можна встановити нижню межу в 5.
- перенести повідомлення про пустий день тижня до поля, замість показувати його для всієї форми
- видалити правило валідації DaysIfNotSchedualed бо воно дуже жорстке. Ми можемо тимчасово зняти екскурсію із розкладу, при цьому не потрібно вимагати від користувача очищувати решту полів

**Comment by thxnk** [comment]
Усе виконано.

**Comment by msenyk** [comment]
@thxnk Все вірно. Невеличка граматична помилка в полі Scheduled (зайва літера "a").
